src/main/java/com/gluonhq/substrate/target/AndroidTargetConfiguration.java

```diff
@@ -407,95 +311,41 @@ private void copyOtherDalvikClasses() throws IOException, InterruptedException {
-        Path androidManifestPath = getApkPath().resolve("AndroidManifest.xml");
-        Path apkResPath = getApkPath().resolve(Constants.ANDROID_RES_FOLDER);
-        ProcessRunner aaptpackage = new ProcessRunner(aaptCmd, "package",
-                "-f", "-m", "-F", unalignedApk,
-                "-M", androidManifestPath.toString(),
-                "-S", apkResPath.toString(),
-                "-I", androidJar);
-        return aaptpackage.runProcess("aaptPackage");
-    }
-
-    private int aaptAddDxClasses(Path buildToolsPath, String unalignedApk) throws IOException, InterruptedException {
-        String aaptCmd = buildToolsPath.resolve("aapt").toString();
-        ProcessRunner aaptAddClass = new ProcessRunner(aaptCmd, "add", unalignedApk, "classes.dex");
-        return aaptAddClass.runProcess("aaptAddDxClasses", getApkBinPath().toFile());
-    }
-
-    private int aaptAddNativeLibs(Path buildToolsPath, String unalignedApk) throws IOException, InterruptedException {
-        String aaptCmd = buildToolsPath.resolve("aapt").toString();
-
-        Path libPath = paths.getAppPath().resolve(getLinkOutputName());
-        Path substrateLibPath = getApkLibArm64Path().resolve("libsubstrate.so");
-        Files.copy(libPath, substrateLibPath, StandardCopyOption.REPLACE_EXISTING);
-
-        List<String> aaptAddLibsArgs = new ArrayList<>(Arrays.asList(aaptCmd, "add", unalignedApk, "lib/arm64-v8a/libsubstrate.so"));
-
         if (projectConfiguration.isUseJavaFX()) {
             Path javafxFreetypeLibPath = fileDeps.getJavaFXSDKLibsPath().resolve("libfreetype.so");
-            Path freetypeLibPath = getApkLibArm64Path().resolve("libfreetype.so");
+            Path freetypeLibPath = projectLibsLocation.resolve("libfreetype.so");
             Files.copy(javafxFreetypeLibPath, freetypeLibPath, StandardCopyOption.REPLACE_EXISTING);
-            aaptAddLibsArgs.add("lib/arm64-v8a/libfreetype.so");
         }
 
-        ProcessRunner aaptAddLibs = new ProcessRunner(aaptAddLibsArgs.toArray(String[]::new));
-        return aaptAddLibs.runProcess("aaptAddNativeLibs", getApkPath().toFile());
+        Path libsubstrate = paths.getAppPath().resolve(getLinkOutputName());
+        Files.copy(libsubstrate, projectLibsLocation.resolve("libsubstrate.so"), StandardCopyOption.REPLACE_EXISTING);
     }
 
-    private int zipAlign(Path buildToolsPath, String unalignedApk, String alignedApk) throws IOException, InterruptedException {
-        String zipAlignCmd = buildToolsPath.resolve("zipalign").toString();
-        ProcessRunner zipAlign = new ProcessRunner(zipAlignCmd, "-f", "4", unalignedApk, alignedApk);
-        return zipAlign.runProcess("zipalign");
-    }
+    /*
+     * Generates release signing configuration if
+     * keystore is provided
+     */
+    private void generateSigningConfiguration() throws IOException {
+        Path settingsFile = getAndroidProjectPath().resolve("keystore.properties");
```

Comment:
The file is not created, is it?

Reply:
It is copied from here
https://github.com/gluonhq/substrate/blob/12fe324a2336c5cbea8aefb29ddcf834001e012a/src/main/resources/native/android/android_project/keystore.properties#L1-L4

Reply:
I saw that, but what I meant is that the file is not created under `gensrc/android_project`. 

Also I don't see where we call release, all the tasks are `assembleDebug` or `installDebug`, and the generated apk is under `outputs/apk/debug/apk-debug`.

Reply:
That file is copied to `/gvm/android_project/keystore.properties`. I don't think it should be in `gensrc`, since `keystore.properties` is auto-generated from `ReleaseConfiguration`, on each `package` task.

I don't call release since `./gradlew assembleRelease` is a lot slower, requires release keystore and isn't required for debugging. I believe that `client:publish` task or something similar might be better path moving forward.

Reply:
Hmm, I still have `gensrc/android_project`, so I've must missed one of your latest commits (hard to say since you squash them ;) ). Will checkout again.

About `publish`, precisely we have added `releaseConfiguration` to the plugin: if you have it, you want to release, so `package` should do that.

Reply:
I added check if `releaseConfiguration` is set to switch between Debug and Release

Reply:
Ok, works now under gvm. However, I still think we should use `assembleRelease` if developer provides `releaseConfiguration` and calls `client:package`.

In order to test it, I'm using `assembleRelease`, and I got an exception:
```
Execution failed for task ':app:validateSigningRelease'.
> Keystore file not set for signing config release
```
I've done some changes:
- moved keystore.properties to `app`, and change the app's build to:

```
release {
            if (file("keystore.properties").exists()) {
                Properties keystoreSettings = new Properties()
                keystoreSettings.load(new FileInputStream(file("keystore.properties")))

                storeFile = file(keystoreSettings['storeFile'])
                storePassword = keystoreSettings['storePassword']
                keyAlias = keystoreSettings['keyAlias']
                keyPassword = keystoreSettings['keyPassword']
            }
        }

buildTypes {
        release {
            signingConfig signingConfigs.release
        }
    }

```

This works fine now (and it takes a few more seconds).

Reply:
I've added those changes in latest commit.
As for `client:package`, https://github.com/gluonhq/substrate/blob/bf585c51d5fbb9eaec6b1bfae86eb9e59c6bf90f/src/main/java/com/gluonhq/substrate/target/AndroidTargetConfiguration.java#L367-L374 should be false if `releaseConfiguration` is provided, so "Release" should be returned and "assembleRelease" emitted.